src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -39,10 +39,10 @@ private static bool ProxySupportsConnectionAuth(HttpResponseMessage response)
             return false;
         }
 
-        private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMessage request, Uri authUri, ICredentials credentials, bool isProxyAuth, HttpConnection connection, HttpConnectionPool connectionPool, CancellationToken cancellationToken)
+        private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMessage request, Uri authUri, ICredentials credentials, bool isProxyAuth, HttpConnection? connection, HttpConnectionPool connectionPool, CancellationToken cancellationToken)
```

Comment:
`HttpConnection? connection` is not null here, but because of the call below on row 67 returns nullable connection and I was not able to use `!` for that expression i make it here nullable
```csharp
(connection, response) = await connectionPool.CreateHttp11ConnectionAsync(request, cancellationToken).ConfigureAwait(false);
```

Reply:
gonna disable warning instead of making `connection and response` nullable

Reply:
> gonna disable warning instead of making connection and response nullable

Sounds fine.  The CreateHttp11ConnectionAsync method being called returns a discrimated union, and it does so in a way that's not particularly friendly for nullable reference types.  The only alternatives I see would be to change that scheme, or to introduce additional locals, but neither of those seems worth it.